lib/perfdata/influxdbchecktask.cpp

```diff
@@ -0,0 +1,156 @@
+
+	double pendingDataBuffersCrit = MacroProcessor::ResolveMacros("$influxdbwriter_pending_queries_critical$", resolvers, checkable->GetLastCheckResult(),
+		&missingPendingDataBuffersCrit, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	/*
+	 * Don't execute built-in commands on the master in remote command execution mode
+	 * see #2267
+	 */
+	if (resolvedMacros && !useResolvedMacros) {
+		return;
+	}
+
+	if (influxdbWriterName.IsEmpty()) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, "Attribute 'influxdbwriter_name' must be set.");
+		return;
+	}
+
+	InfluxdbWriter::Ptr influxdbWriter = InfluxdbWriter::GetByName(influxdbWriterName);
+
+	if (!influxdbWriter) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, "InfluxdbWriter '" + influxdbWriterName + "' object does not exist.");
+		return;
+	}
+
+	if (influxdbWriter->IsPaused()) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, "InfluxdbWriter is temporarily disabled on this cluster instance.", state);
+		return;
+	}
+
+	double wqItemPerSec = influxdbWriter->GetQueryCount(60) / 60.0;
+	double pendingDataBuffers = influxdbWriter->GetPendingDataBuffers();
+	std::ostringstream msgBuffer;
+
+	msgBuffer << "Queries per second: " << std::fixed << std::setprecision(3) << wqItemPerSec << ".";
+
+	/* Check whether the thresholds have been defined and match. */
+	if (missingQueriesCritical.IsEmpty() && wqItemPerSec < queriesCritical) {
+		msgBuffer << " " << wqItemPerSec << " queries/s lower than critical threshold (" << queriesCritical << " queries/s).";
+
+		state = ServiceCritical;
+	} else if (missingQueriesWarning.IsEmpty() && wqItemPerSec < queriesWarning) {
+		msgBuffer << " " << wqItemPerSec << " queries/s lower than warning threshold (" << queriesWarning << " queries/s).";
+
+		state = ServiceWarning;
+	}
+
+	if (missingPendingDataBuffersCrit.IsEmpty() && pendingDataBuffers > pendingDataBuffersCrit) {
+		msgBuffer << " " << pendingDataBuffers << " pending data buffers greater than critical threshold ("
+			<< pendingDataBuffersCrit << " queries).";
+
+		state = ServiceCritical;
+	} else if (missingPendingDataBuffersWarn.IsEmpty() && pendingDataBuffers > pendingDataBuffersWarn) {
+		msgBuffer << " " << pendingDataBuffers << " pending data buffers greater than warning threshold ("
+			<< pendingDataBuffersWarn << " queries).";
+
+		if (state == ServiceOK) {
+			state = ServiceWarning;
+		}
+	}
+
+	cr->SetPerformanceData(new Array({
+		{ new PerfdataValue("queries", wqItemPerSec, false, "", queriesWarning, queriesCritical) },
+		{ new PerfdataValue("queries_1min", influxdbWriter->GetQueryCount(60)) },
+		{ new PerfdataValue("queries_5mins", influxdbWriter->GetQueryCount(5 * 60)) },
+		{ new PerfdataValue("queries_15mins", influxdbWriter->GetQueryCount(15 * 60)) },
+		{ new PerfdataValue("pending_datas", pendingDataBuffers, false, "", pendingDataBuffersWarn, pendingDataBuffersCrit) }
```

Comment:
This should probably be called `pending_queries` or `pending_writes` (not sure if InfluxDB has any specific terminology there, I'd just go for queries as that's used for the other values already). Variable and method should be renamed accordingly.